auth/downscoping/snippets.py

```diff
@@ -0,0 +1,184 @@
+    # [END auth_downscoping_initialize_downscoped_cred]
+    return (access_token, expiry)
+
+
+# [END auth_downscoping_token_broker]
+
+
+# [START auth_downscoping_token_consumer]
+def token_consumer(bucket_name, object_name):
+    """Tests token consumer readonly access to the specified object.
+
+    Args:
+        bucket_name (str): The name of the cloud storage bucket.
+        object_name (str): The name of the object in the storage bucket to
+            read.
+    """
+    # Create the OAuth credentials from the downscoped token and pass a
+    # refresh handler to handle token expiration. Passing the original
+    # downscoped token or the expiry here is optional, as the refresh_handler
+    # will generate the downscoped token on demand.
+    def refresh_handler(request, scopes=None):
+        # The common pattern of usage is to have a token broker pass the
+        # downscoped short-lived access tokens to a token consumer via some
+        # secure authenticated channel.
+        # For illustration purposes, we are generating the downscoped token
+        # locally.
+        return get_token_from_broker(bucket_name, OBJECT_PREFIX_NAME)
+
+    creds = credentials.Credentials(
+        None,
+        scopes=["https://www.googleapis.com/auth/cloud-platform"],
+        refresh_handler=refresh_handler,
+    )
+
+    # Initialize a storage client with the oauth2 credentials.
+    storage_client = storage.Client(credentials=creds)
+    # The token broker has readonly access to the specified bucket object.
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(object_name)
+    print(blob.download_as_bytes().decode("utf-8"))
+
+
+# [END auth_downscoping_token_consumer]
+
+
+def upload_object(storage_client, bucket_name, source_file_name, destination_blob_name):
+    """Upload object from source file to specified object under blob name.
+
+    By default the object is created as non-public.
+
+    Args:
+        storage_client (google.cloud.storage.Client): The storage client
+            instance.
+        bucket_name (str): The name of the cloud storage bucket.
+        source_file_name (str): The file path to the source file to upload.
+        destination_blob_name (str): The object name under which the file will
+            be uploaded.
+    """
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_filename(source_file_name)
+
+
+def delete_object(storage_client, bucket_name, blob_name):
+    """Delete object of blob name under specified bucket.
+
+    Args:
+        storage_client (google.cloud.storage.Client): The storage client
+            instance.
+        bucket_name (str): The name of the cloud storage bucket.
+        blob_name (str): The object name in the storage bucket to be deleted.
+    """
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(blob_name)
+    blob.delete()
+
+
+def main(bucket_name, filename):
```

Comment:
Note: samples do not have to show how to setup/teardown all the resources they need.  As an example, [this sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/33e3de913c30d430acc040094264b36142c04d89/storage/s3-sdk/list_gcs_objects.py) is focused on showing how to list objects in a blob, so it _doesn't_ show how to create a storage bucket and add objects to it.

If having the sample be runnable via a `main()` is by design please disregard. :)

Reply:
I moved the setup to the test.